ocaml/typing/uniqueness_analysis.ml

```diff
@@ -726,19 +1316,53 @@ end = struct
 
   let par t0 t1 = map2 Usage_tree.par t0 t1
 
-  let chooses l = List.fold_left choose unused l
+  (* For convenience, our semirings do not have a zero.
+     However, when we fold the [choose] operation we need to initialize the
+     accumulator with zero. We avoid doing this here by skipping the initial
+     accumulator altogether. When [chooses] gets passed an empty list,
+     we return 'empty' which is sound in all semirings with '0 > 1'. *)
```

Comment:
This comment could use a little elaboration. Understanding it requires knowing that `empty` is `1` in the semiring. And: do all semirings here have `0 > 1`? The comment doesn't say.